test/unit/client_interface_sections/search_test.rb

```diff
@@ -18,7 +26,137 @@ def test_url_encoding_only_happens_once
         }
       }
     )
-    assert_equal 'search-test/Appointment?date=%3E2016-01-01&patient=test',
+    assert_equal 'http://search-test/Appointment?date=%3E2016-01-01&patient=test',
+                 reply.request[:url]
+  end
+
+  def test_get_search
+    stub_request(:get, 'http://search-test/Patient?address=123%20Sesame%20Street&given=name').
+      to_return(empty_search_response)
+
+    reply = @client.search(
+      FHIR::Patient,
+      {
+        search: {
+          parameters: {
+            given: 'name',
+            address: '123 Sesame Street'
+          }
+        }
+      }
+    )
+    assert_equal 'http://search-test/Patient?address=123%20Sesame%20Street&given=name',
```

Comment:
It makes sense to make an assertion on the actual url that was requested in the test for double encoding, but for these others I'd do something like this:
```
search = stub_request(...)
...
assert_requested(search)
```

Reply:
Fixed!